py_ecc/secp256k1/secp256k1.py

```diff
@@ -137,7 +137,7 @@ def ecdsa_raw_sign(msghash, priv):
     return v, r, s
 
 
-def ecdsa_raw_recover(msghash, vrs):
+def ecdsa_raw_recover(msghash: str, vrs: Tuple[int, int, int]) -> Union[bool, Tuple[int, int]]:
```

Comment:
It'd be nice to refactor this so to replace `return False` with an exception so that the return type can be more firm here.  Is that something you'd feel comfortable doing?

Reply:
sure @pipermerriam, I like to add 
```
if (xcubedaxb - y * y) % P != 0 or not (r % N) or not (s % N):
	return ValueError("sig is invalid, %d cannot be the x coord for point on curve" % r)
```